src/main/scala/org/scalameter/Executor.scala

```diff
@@ -1,18 +1,21 @@
 package org.scalameter
 
 
-import collection._
-import compat._
-import utils.{withGCNotification, Tree}
+import utils.Tree
 
 
 
 trait Executor {
 
+  def measurer: Measurer
+
   def run[T](setuptree: Tree[Setup[T]], reporter: Reporter, persistor: Persistor): Tree[CurveData] = {
+    val context = measurer.prepareContext(setuptree.context)
```

Comment:
It seems to me that the initial `prepareContext` call must:
a) do some side-effecting actions to prepare a testing environment
b) return a couple of values that describe the prepared testing environment

# Alternative A

Initially, I was hoping we could replace the top-level context altogether before it gets used to create `Tree[Setup[T]]`, and that `BasePerformanceTest#defaultConfig` could be used for this purpose. Now I see that this is broken - `defaultConfig` is just ignored.

The way to fix it is to use `defaultConfig` in `BasePerformanceTest` to initialize the `setupzipper` dynamic variable with the correct context.
Something like this should do it:

    def defaultConfig: 


# Alternative B